tests/PHPStan/Analyser/nsrt/bug-11716.php

```diff
@@ -136,3 +136,16 @@ function emptyArrr($mixed, array $arr)
     }
     assertType('mixed', $mixed);
 }
+
+function emptyString($mixed)
+{
+    // see https://3v4l.org/XHZdr
+    $arr = ['' => 1, 'a' => 2];
+
+    if (isset($arr[$mixed])) {
+        assertType("''|'a'|null", $mixed);
+    } else {
+        assertType('mixed', $mixed);
```

Comment:
`mixed~(''|'a'|null)`

Reply:
I guess this comment went on the wrong line (or I have puhsed a change at the same time you did the comment) and therefore it does not make sense atm?

Reply:
I see it on correct line, it is for falsy context:
```
    $arr = [0 => 0, 1 => 1, 2 => 2];
    if (isset($arr[$mixed])) {
    } else {
        // falsy context
    }
```

if `$arr[$k]` is set (and not `null` because of `isset`), then the consts types in truthy context can be substracted in the falsy context

Reply:
This is sealed vs unsealed array territory. I would not do this until we have a clear separation of the 2 types

Reply:
Really? If array specifies some non-optional keys, they must be present (in sealed and unsealed array) and can be substracted in the else branch, can't they?

Reply:
looking at it again, I think you are right. 

I still don't want to implement that because the PR is already big enough, `isset()` handling already is pretty complicated, and I think its pretty unlikely this falsy-context narrowing would be useful in practice.

Reply:
```!isset($arr[$k])``` is quite common code pattern, can it be impl. at least for constant types excl. float?

Reply:
Please open a new issue with test expectations